examples/kotlin/src/main/kotlin/com/example/booktest/postgresql/QueriesImpl.kt

```diff
@@ -275,31 +238,24 @@ class QueriesImpl(private val conn: Connection) : Queries {
                 results.getObject(7, OffsetDateTime::class.java),
                 (results.getArray(8).array as Array<String>).toList()
             )
-          if (results.next()) {
-              throw SQLException("expected one row in result set, but got many")
-          }
-          ret
-        }
+      if (results.next()) {
+          throw SQLException("expected one row in result set, but got many")
```

Comment:
Do you support a `Book?` type such that the generated code can return a `null` instead of throwing an exception?

Reply:
So that's an interesting question. In Go one usually would get a `sql.ErrNoRows` when querying for a single row but the resultset is empty. In Kotlin this throws an exception that needs to be caught. Except there's no such canonical exception (I think), so we need to create one and stick it in a runtime library.

Or, maybe the `:one` and `:many` commands for sqlc which return 1 row or a list of rows respectively, should actually continue to return a `RowResults` and `ListResults` class? The `RowResults` could have `.get()` which throws and `getOrNull()` which does not. While the `ListResults` class could have a `getList()` to get the full list or `next()` to get the next item in the list.

Reply:
Oooo or return an `Iterator()` 😱